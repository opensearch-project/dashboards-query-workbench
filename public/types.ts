/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DataSourcePluginStart } from '../../../src/plugins/data_source/public/types';
import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';
import { ObservabilityStart } from './types';

export interface WorkbenchPluginSetup {}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WorkbenchPluginStart {
}

export interface AppPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
  dataSource: DataSourcePluginStart;
  observabilityDashboards?: ObservabilityStart;
}

export type { ObservabilityStart } from '../../dashboards-observability/public';
