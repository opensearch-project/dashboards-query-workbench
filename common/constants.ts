/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */


export const ON_LOAD_QUERY = `SHOW tables LIKE '%';`
