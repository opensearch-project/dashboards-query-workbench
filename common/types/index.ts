/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export type AggregationFunctionType = 'count' | 'sum' | 'avg' | 'max' | 'min';

export interface MaterializedViewColumn {
  id: string;
  functionName: AggregationFunctionType;
  functionParam: string;
  fieldAlias?: string;
}

export type SkippingIndexAccMethodType = 'PARTITION' | 'VALUE_SET' | 'MIN_MAX';

export interface SkippingIndexRowType {
  id: string;
  fieldName: string;
  dataType: string;
  accelerationMethod: SkippingIndexAccMethodType;
}

export interface DataTableFieldsType {
  id: string;
  fieldName: string;
  dataType: string;
}

export interface RefreshIntervalType {
  refreshWindow: number;
  refreshInterval: string;
}

export type AccelerationIndexType = 'skipping' | 'covering' | 'materialized';

export interface GroupByTumbleType {
  timeField: string;
  tumbleWindow: number;
  tumbleInterval: string;
}

export interface materializedViewQueryType {
  columnsValues: MaterializedViewColumn[];
  groupByTumbleValue: GroupByTumbleType;
}

export interface FormErrorsType {
  dataSourceError: string[];
  databaseError: string[];
  dataTableError: string[];
  skippingIndexError: string[];
  coveringIndexError: string[];
  materializedViewError: string[];
  indexNameError: string[];
  primaryShardsError: string[];
  replicaShardsError: string[];
  refreshIntervalError: string[];
  checkpointLocationError: string[];
}

export type AccelerationRefreshType = 'auto' | 'interval' | 'manual';

export interface CreateAccelerationForm {
  dataSource: string;
  database: string;
  dataTable: string;
  dataTableFields: DataTableFieldsType[];
  accelerationIndexType: AccelerationIndexType;
  skippingIndexQueryData: SkippingIndexRowType[];
  coveringIndexQueryData: string[];
  materializedViewQueryData: materializedViewQueryType;
  accelerationIndexName: string;
  primaryShardsCount: number;
  replicaShardsCount: number;
  refreshType: AccelerationRefreshType;
  checkpointLocation: string | undefined;
  refreshIntervalOptions: RefreshIntervalType;
  formErrors: FormErrorsType;
}

export type AsyncQueryLoadingStatus = 'SUCCESS' | 'FAILED' | 'RUNNING' | 'SCHEDULED' | 'CANCELLED';
export type TreeItemType = 'covering_index' | 'skipping_index' | 'table' | 'database' | 'materialized_view' | 'Load Materialized View' | 'badge'

export interface TreeItem {
  name: string;
  type: TreeItemType;
  isExpanded: boolean;
  values?: TreeItem[];
  isLoading?: boolean
}

export interface isLoading {
  flag: boolean;
  status: string;
}
