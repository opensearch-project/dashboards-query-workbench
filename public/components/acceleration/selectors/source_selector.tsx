/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiComboBox,
  EuiComboBoxOptionOption,
  EuiFormRow,
  EuiSpacer,
  EuiText,
  htmlIdGenerator,
} from '@elastic/eui';
import React, { useEffect, useState } from 'react';
import { CreateAccelerationForm } from '../../../../common/types';

interface AccelerationDataSourceSelectorProps {
  accelerationFormData: CreateAccelerationForm;
  setAccelerationFormData: React.Dispatch<React.SetStateAction<CreateAccelerationForm>>;
}

export const AccelerationDataSourceSelector = ({
  accelerationFormData,
  setAccelerationFormData,
}: AccelerationDataSourceSelectorProps) => {
  const [dataConnections, setDataConnections] = useState<EuiComboBoxOptionOption<string>[]>([]);
  const [selectedDataConnection, setSelectedDataConnection] = useState<
    EuiComboBoxOptionOption<string>[]
  >([]);
  const [databases, setDatabases] = useState<EuiComboBoxOptionOption<string>[]>([]);
  const [selectedDatabase, setSelectedDatabase] = useState<EuiComboBoxOptionOption<string>[]>([]);
  const [tables, setTables] = useState<EuiComboBoxOptionOption<string>[]>([]);
  const [selectedTable, setSelectedTable] = useState<EuiComboBoxOptionOption<string>[]>([]);

  useEffect(() => {
    // TODO: remove hardcoded responses
    setDataConnections([
      {
        label: 'spark1',
      },
      {
        label: 'spark2',
      },
    ]);
  }, []);

  useEffect(() => {
    // TODO: remove hardcoded responses
    if (accelerationFormData.dataSource !== '') {
      setDatabases([
        {
          label: 'Database1',
        },
        {
          label: 'Database2',
        },
      ]);
    }
  }, [accelerationFormData.dataSource]);

  useEffect(() => {
    // TODO: remove hardcoded responses
    if (accelerationFormData.database !== '') {
      setTables([
        {
          label: 'Table1',
        },
        {
          label: 'Table2',
        },
      ]);
    }
  }, [accelerationFormData.database]);

  useEffect(() => {
    // TODO: remove hardcoded responses
    if (accelerationFormData.dataTable !== '') {
      const idPrefix = htmlIdGenerator()();
      setAccelerationFormData({
        ...accelerationFormData,
        dataTableFields: [
          { id: `${idPrefix}1`, fieldName: 'Field1', dataType: 'Integer' },
          { id: `${idPrefix}2`, fieldName: 'Field2', dataType: 'Integer' },
          { id: `${idPrefix}3`, fieldName: 'Field3', dataType: 'Integer' },
          { id: `${idPrefix}4`, fieldName: 'Field4', dataType: 'Integer' },
          { id: `${idPrefix}5`, fieldName: 'Field5', dataType: 'Integer' },
          { id: `${idPrefix}6`, fieldName: 'Field6', dataType: 'Integer' },
          { id: `${idPrefix}7`, fieldName: 'Field7', dataType: 'Integer' },
          { id: `${idPrefix}8`, fieldName: 'Field8', dataType: 'Integer' },
          { id: `${idPrefix}9`, fieldName: 'Field9', dataType: 'Integer' },
          { id: `${idPrefix}10`, fieldName: 'Field10', dataType: 'Integer' },
          { id: `${idPrefix}11`, fieldName: 'Field11', dataType: 'Integer' },
          { id: `${idPrefix}12`, fieldName: 'Field12', dataType: 'TimestampType' },
        ],
      });
    }
  }, [accelerationFormData.dataTable]);

  return (
    <>
      <EuiText data-test-subj="datasource-selector-header">
        <h3>Select data source</h3>
      </EuiText>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        Select data connection where the data you want to accelerate resides.
      </EuiText>
      <EuiSpacer size="s" />
      <EuiFormRow
        label="Data source"
        helpText="A data source has to be configured and active to be able to select it and index data from."
      >
        <EuiComboBox
          placeholder="Select a data source"
          singleSelection={{ asPlainText: true }}
          options={dataConnections}
          selectedOptions={selectedDataConnection}
          onChange={(dataConnectionOptions) => {
            setAccelerationFormData({
              ...accelerationFormData,
              dataSource: dataConnectionOptions[0].label,
            });
            setSelectedDataConnection(dataConnectionOptions);
          }}
          isInvalid={selectedDataConnection.length === 0}
          isClearable={false}
        />
      </EuiFormRow>
      <EuiFormRow
        label="Database"
        helpText="Select the database that contains the tables you'd like to use."
      >
        <EuiComboBox
          placeholder="Select a database"
          singleSelection={{ asPlainText: true }}
          options={databases}
          selectedOptions={selectedDatabase}
          onChange={(tableOptions) => {
            setAccelerationFormData({
              ...accelerationFormData,
              database: tableOptions[0].label,
            });
            setSelectedDatabase(tableOptions);
          }}
          isInvalid={selectedDatabase.length === 0}
          isClearable={false}
        />
      </EuiFormRow>
      <EuiFormRow
        label="Table"
        helpText="Select the Spark table that has the data you would like to index."
      >
        <EuiComboBox
          placeholder="Select a table"
          singleSelection={{ asPlainText: true }}
          options={tables}
          selectedOptions={selectedTable}
          onChange={(tableOptions) => {
            setAccelerationFormData({
              ...accelerationFormData,
              dataTable: tableOptions[0].label,
            });
            setSelectedTable(tableOptions);
          }}
          isInvalid={selectedTable.length === 0}
          isClearable={false}
        />
      </EuiFormRow>
    </>
  );
};
