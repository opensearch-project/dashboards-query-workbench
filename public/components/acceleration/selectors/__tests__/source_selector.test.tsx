/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiComboBoxOptionOption } from '@elastic/eui';
import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { CreateAccelerationForm } from '../../../../../common/types';
import { httpClientMock } from '../../../../../test/mocks';
import { createAccelerationEmptyDataMock } from '../../../../../test/mocks/accelerationMock';
import { mockDatasourcesQuery } from '../../../../../test/mocks/mockData';
import { AccelerationDataSourceSelector } from '../source_selector';

describe('Source selector components', () => {
  configure({ adapter: new Adapter() });

  it('renders source selector with default options', async () => {
    const accelerationFormData = createAccelerationEmptyDataMock;
    const selectedDatasource: EuiComboBoxOptionOption[] = [];
    const setAccelerationFormData = jest.fn();
    const client = httpClientMock;
    client.get = jest.fn().mockResolvedValue(mockDatasourcesQuery);

    const wrapper = mount(
      <AccelerationDataSourceSelector
        http={client}
        selectedDatasource={selectedDatasource}
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders source selector with different options', async () => {
    const selectedDatasource: EuiComboBoxOptionOption[] = [{ label: 'ds' }];
    const accelerationFormData: CreateAccelerationForm = {
      ...createAccelerationEmptyDataMock,
      dataSource: 'ds',
      database: 'db',
      dataTable: 'tb',
    };
    const setAccelerationFormData = jest.fn();
    const client = httpClientMock;
    client.get = jest.fn().mockResolvedValue(mockDatasourcesQuery);
    client.post = jest.fn().mockResolvedValue([]);
    const wrapper = mount(
      <AccelerationDataSourceSelector
        selectedDatasource={selectedDatasource}
        http={client}
        accelerationFormData={accelerationFormData}
        setAccelerationFormData={setAccelerationFormData}
      />
    );
    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          noKey: false,
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
