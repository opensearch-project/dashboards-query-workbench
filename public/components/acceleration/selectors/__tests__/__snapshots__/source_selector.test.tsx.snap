// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source selector components renders source selector with default options 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="datasource-selector-header"
  >
    <h3>
      Select data source
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      Select data connection where the data you want to accelerate resides.
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Data source
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a data source
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        A data source has to be configured and active to be able to select it and index data from.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Database
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a database
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the database that contains the tables you'd like to use.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Table
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a table
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the Spark table that has the data you would like to index.
      </div>
    </div>
  </div>,
]
`;

exports[`Source selector components renders source selector with different options 1`] = `
Array [
  <div
    className="euiText euiText--medium"
    data-test-subj="datasource-selector-header"
  >
    <h3>
      Select data source
    </h3>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      Select data connection where the data you want to accelerate resides.
    </div>
  </div>,
  <div
    className="euiSpacer euiSpacer--s"
  />,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Data source
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <span
                className="euiComboBoxPill euiComboBoxPill--plainText"
              >
                ds
              </span>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        A data source has to be configured and active to be able to select it and index data from.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Database
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a database
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the database that contains the tables you'd like to use.
      </div>
    </div>
  </div>,
  <div
    className="euiFormRow"
    id="some_html_id-row"
  >
    <div
      className="euiFormRow__labelWrapper"
    >
      <label
        aria-invalid={false}
        className="euiFormLabel euiFormRow__label"
        htmlFor="some_html_id"
      >
        Table
      </label>
    </div>
    <div
      className="euiFormRow__fieldWrapper"
    >
      <div
        aria-describedby="some_html_id-help-0"
        aria-expanded={false}
        aria-haspopup="listbox"
        className="euiComboBox"
        onKeyDown={[Function]}
        role="combobox"
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <div
              className="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isLoading"
              data-test-subj="comboBoxInput"
              onClick={[Function]}
              tabIndex={-1}
            >
              <p
                className="euiComboBoxPlaceholder"
              >
                Select a table
              </p>
              <div
                className="euiComboBox__input"
                style={
                  Object {
                    "display": "inline-block",
                    "fontSize": 14,
                  }
                }
              >
                <input
                  aria-controls=""
                  data-test-subj="comboBoxSearchInput"
                  id="some_html_id"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "2px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
            >
              <span
                className="euiLoadingSpinner euiLoadingSpinner--medium"
              />
              <button
                aria-label="Open list of options"
                className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                data-test-subj="comboBoxToggleListButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                  focusable="false"
                  height={16}
                  role="img"
                  style={null}
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                    fillRule="non-zero"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormHelpText euiFormRow__text"
        id="some_html_id-help-0"
      >
        Select the Spark table that has the data you would like to index.
      </div>
    </div>
  </div>,
]
`;
